game/game/gamescript.cpp

```diff
@@ -957,125 +935,123 @@ int GameScript::invokeState(Npc* npc, Npc* oth, Npc* vic, ScriptFn fn) {
-  auto fn = getSymbolIndex(func);
-  if(fn==size_t(-1)) {
+int GameScript::invokeCond(Npc& npc, const std::string& func) {
+  auto fn = vm.find_symbol_by_name(func);
+  if(fn==nullptr) {
     Gothic::inst().onPrint("MOBSI::conditionFunc is not invalid");
     return 1;
     }
-  ScopeVar self(vm, vm.globalSelf(),  npc);
-  return runFunction(fn);
+  ScopeVar self(vm.global_self(), npc.handle());
+  return vm.call_function<int>(fn);
   }
 
 void GameScript::invokePickLock(Npc& npc, int bSuccess, int bBrokenOpen) {
-  auto& dat = vm.getDATFile();
-  auto fn   = dat.getSymbolIndexByName("G_PickLock");
-  if(fn==size_t(-1))
+  auto fn   = vm.find_symbol_by_name("G_PickLock");
+  if(fn==nullptr)
     return;
-  ScopeVar self(vm, vm.globalSelf(),  npc);
-  vm.pushInt(bSuccess);
-  vm.pushInt(bBrokenOpen);
-  runFunction(fn);
+  ScopeVar self(vm.global_self(), npc.handle());
+  vm.call_function<void>(fn, bSuccess, bBrokenOpen);
   }
 
 CollideMask GameScript::canNpcCollideWithSpell(Npc& npc, Npc* shooter, int32_t spellId) {
-  auto& dat = vm.getDATFile();
-  auto fn   = dat.getSymbolIndexByName("C_CanNpcCollideWithSpell");
-  if(fn==size_t(-1))
+  auto fn   = vm.find_symbol_by_name("C_CanNpcCollideWithSpell");
+  if(fn==nullptr)
     return COLL_DOEVERYTHING;
 
-  ScopeVar self (vm, vm.globalSelf(),  npc);
-  ScopeVar other(vm, vm.globalOther(), shooter);
-  vm.pushInt(spellId);
-  int v = runFunction(fn);
-  return CollideMask(v);
+  ScopeVar self (vm.global_self(),  npc.handle());
+  ScopeVar other(vm.global_other(), shooter->handle());
+  return CollideMask(vm.call_function<int>(fn, spellId));
   }
 
 int GameScript::playerHotKeyScreenMap(Npc& pl) {
-  auto& dat = vm.getDATFile();
-  auto fn   = dat.getSymbolIndexByName("player_hotkey_screen_map");
-  if(fn==size_t(-1))
+  auto fn   = vm.find_symbol_by_name("player_hotkey_screen_map");
+  if(fn==nullptr)
     return -1;
 
-  ScopeVar self(vm, vm.globalSelf(), pl);
-  int map = runFunction(fn);
+  ScopeVar self(vm.global_self(), pl.handle());
+  int map = vm.call_function<int>(fn);
   if(map>=0)
     pl.useItem(size_t(map));
   return map;
   }
 
-const Daedalus::ZString& GameScript::spellCastAnim(Npc&, Item &it) {
-  if(spellFxAniLetters==size_t(-1)) {
-    static const Daedalus::ZString FIB = "FIB";
+const std::string& GameScript::spellCastAnim(Npc&, Item &it) {
+  if(spellFxAniLetters==nullptr) {
+    static const std::string FIB = "FIB";
     return FIB;
     }
-  auto& spellAni = vm.getDATFile().getSymbolByIndex(spellFxAniLetters);
-  auto& tag      = spellAni.getString(size_t(it.spellId()));
-  return tag;
+  return spellFxAniLetters->get_string(it.spellId());
   }
 
-bool GameScript::aiOutput(Npc &npc, const Daedalus::ZString& outputname, bool overlay) {
+bool GameScript::aiOutput(Npc &npc, std::string_view outputname, bool overlay) {
   char buf[256]={};
-  std::snprintf(buf,sizeof(buf),"%s.WAV",outputname.c_str());
+  std::snprintf(buf,sizeof(buf),"%s.WAV",outputname.data());
```

Comment:
It's not valid (strictly speaking) to print `.data()`, since string_view has no null-termination symbol.
`std::snprintf(buf,sizeof(buf),"%.*s.WAV",int(outputname.size()),outputname.data()))`